File: frontend/src/stores/survivor.ts
import { defineStore } from "pinia";
import { ref } from "vue";
import { useApi } from "@/composables/useApi";

export interface SurvivorPick {
  matchday_number: number;
  team: string;
  match_id: string;
  result: string; // pending | won | lost | draw
}

export interface SurvivorEntry {
  id: string;
  status: string; // alive | eliminated | not_started
  picks: SurvivorPick[];
  used_teams: string[];
  streak: number;
  eliminated_at: string | null;
}

export interface SurvivorStanding {
  user_id: string;
  alias: string;
  status: string;
  streak: number;
  eliminated_at: string | null;
}

export const useSurvivorStore = defineStore("survivor", () => {
  const api = useApi();

  const entry = ref<SurvivorEntry | null>(null);
  const standings = ref<SurvivorStanding[]>([]);
  const loading = ref(false);

  async function fetchStatus(squadId: string, sport: string, season?: number) {
    loading.value = true;
    try {
      const params: Record<string, string> = { sport };
      if (season) params.season = String(season);
      entry.value = await api.get<SurvivorEntry>(`/survivor/${squadId}/status`, params);
    } catch {
      entry.value = null;
    } finally {
      loading.value = false;
    }
  }

  async function makePick(squadId: string, matchId: string, team: string) {
    const result = await api.post<SurvivorEntry>(`/survivor/${squadId}/pick`, {
      match_id: matchId,
      team,
    });
    entry.value = result;
    return result;
  }

  async function fetchStandings(squadId: string, sport: string, season?: number) {
    try {
      const params: Record<string, string> = { sport };
      if (season) params.season = String(season);
      standings.value = await api.get<SurvivorStanding[]>(
        `/survivor/${squadId}/standings`,
        params,
      );
    } catch {
      standings.value = [];
    }
  }

  function reset() {
    entry.value = null;
    standings.value = [];
  }

  return {
    entry,
    standings,
    loading,
    fetchStatus,
    makePick,
    fetchStandings,
    reset,
  };
});

File: backend/app/models/survivor.py
"""Survivor mode models — pick one team per matchday, eliminated on loss."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class SurvivorPick(BaseModel):
    matchday_number: int
    team: str
    team_name: Optional[str] = None
    team_id: Optional[str] = None
    match_id: str
    result: str = "pending"  # pending | won | lost | draw


class SurvivorEntryInDB(BaseModel):
    """One entry per user per squad per sport per season."""
    user_id: str
    squad_id: str
    league_id: int
    season: int
    status: str = "alive"  # alive | eliminated
    picks: list[SurvivorPick] = []
    used_teams: list[str] = []
    used_team_ids: list[str] = []
    streak: int = 0
    eliminated_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class SurvivorPickCreate(BaseModel):
    """Request body for making a survivor pick."""
    match_id: str
    team: str  # team name to pick


class SurvivorEntryResponse(BaseModel):
    """Survivor status returned to the client."""
    id: str
    status: str
    picks: list[dict]
    used_teams: list[str]
    used_team_ids: list[str] = []
    streak: int
    eliminated_at: Optional[datetime] = None


class SurvivorStandingEntry(BaseModel):
    """Single entry in survivor standings."""
    user_id: str
    alias: str
    status: str
    streak: int
    last_pick: Optional[str] = None
    eliminated_at: Optional[datetime] = None
